lib/protobuf/field/base_field_method_definitions.rb

```diff
@@ -0,0 +1,371 @@
+module Protobuf
+  module Field
+    class BaseFieldMethodDefinitions
+
+      def self.fully_qualified_name_string(selph)
+        fully_qualified_name = ""
+        fully_qualified_name << ":"
+        fully_qualified_name << '"' if selph.fully_qualified_name.to_s.start_with?(".")
+        fully_qualified_name << selph.fully_qualified_name.to_s
+        fully_qualified_name << '"' if selph.fully_qualified_name.to_s.start_with?(".")
+        fully_qualified_name
+      end
+
+      def self.define_to_hash_value_to_message_hash!(selph)
```

Comment:
I meant to say "**not** an instance of...". 😞 Updated the comment.

So you are saying that it is a reference to the field class that the methods are being added to?

Reply:
yes, it's `self` in terms of the object he methods are being defined on; but they are singleton methods, we can't apply the methods to the Class because they behave differently based on the instance of the object